src/sensr_client.cpp

```diff
@@ -1,47 +1,133 @@
 #include "sensr_client.h"
-
-const int MAX_ADDRESS_LEN = 512;
+#include "websocket_endpoint.h"
+#include "sensr_message_listener.h"
+#include <algorithm>
 
 namespace sensr
 {
-Client::Client(const char *address)
-    : context_(), subscriber_(context_, ZMQ_SUB)
-{
-  char buffer[MAX_ADDRESS_LEN];
-  int result = snprintf(&buffer[0], MAX_ADDRESS_LEN, "tcp://%s:5050", address);
-  if( result > 0 )
-  { 
-    subscriber_.connect(buffer);
-    subscriber_.setsockopt(ZMQ_SUBSCRIBE, "", 0);
+  Client::Client(const std::string& address) : address_(address) {
+    output_endpoint_.reset(new WebSocketEndPoint());
```

Comment:
Is this possible?
```suggestion
    output_endpoint_ = std::make_unique<WebSocketEndPoint>();
```

Reply:
We're using C++11 in the sensr sdk according to the cmake, so it is not available :feelsbadman:

Reply:
Gotcha :feelsbadman:

Reply:
Sorry, we still use c++11 in SDK. I think lower version can be more attractive for some customers maybe :question:

Reply:
Yeah, right. Due to the compatibility with customers' code, we cannot simply upgrade the standard.